bpf/bpf_host.c

```diff
@@ -845,27 +846,27 @@ do_netdev(struct __ctx_buff *ctx, __u16 proto, const bool from_host)
 	int ret;
 
 #ifdef ENABLE_IPSEC
-	if (from_host) {
-		__u32 magic = ctx->mark & MARK_MAGIC_HOST_MASK;
```

Comment:
we dropped the masking with `MARK_MAGIC_HOST_MASK`. can that lead to issues if `ctx->mark` has other bits set and we're using `==` instead of `&` to check if something is set?

Reply:
In this case, I think it won't happen, because, after the change, the magic value will come from `inherit_identity_from_host`. Inside it, it does `ctx->mark & MARK_MAGIC_HOST_MASK`.

https://github.com/cilium/cilium/pull/18608/files#diff-ad5e0368477133b286664dda204c86e3f6859fe826e0638f9f961280153d8f7cR60

Reply:
oh right. sloppy reading on my part. lgtm!